sztp-agent/pkg/secureAgent/daemon.go

```diff
@@ -186,3 +195,65 @@ func (a *Agent) downloadAndValidateImage() error {
 	}
 	return nil
 }
+
+func (a *Agent) launchPreConfiguration() error {
+	log.Println("[INFO] Starting the Pre-Configuration.")
+	file, err := os.Create(ARTIFACTS_PATH + a.BootstrapServerOnboardingInfo.IetfSztpConveyedInfoOnboardingInformation.InfoTimestampReference + "preconfiguration.sh")
+	if err != nil {
+		log.Println("[ERROR] creating the  pre-configuration script", err.Error())
+		return err
+	}
+	defer file.Close()
+	plainTest, err := base64.StdEncoding.DecodeString(a.BootstrapServerOnboardingInfo.IetfSztpConveyedInfoOnboardingInformation.PreConfigurationScript)
+	_, err = file.WriteString(string(plainTest))
+	if err != nil {
+		log.Println("[ERROR] writing the  pre-configuration script", err.Error())
+		return err
+	}
+	err = os.Chmod(ARTIFACTS_PATH+a.BootstrapServerOnboardingInfo.IetfSztpConveyedInfoOnboardingInformation.InfoTimestampReference+"preconfiguration.sh", 0755)
+	if err != nil {
+		log.Println("[ERROR] changing the  pre-configuration script permission", err.Error())
+		return err
+	}
+	log.Println("[INFO] Pre-Configuration script created successfully")
+	cmd := exec.Command("/bin/sh", ARTIFACTS_PATH+a.BootstrapServerOnboardingInfo.IetfSztpConveyedInfoOnboardingInformation.InfoTimestampReference+"preconfiguration.sh")
+	out, err := cmd.Output()
+	if err != nil {
+		log.Println("[ERROR] running the  pre-configuration script", err.Error())
+		return err
+	}
+	log.Println(string(out)) //remove it
+	log.Println("[INFO] Pre-Configuration script executed successfully")
+	return nil
+}
+
+func (a *Agent) launchPostConfiguration() error {
```

Comment:
consider reducing code duplication and create a function here that both `pre` and `post` call

Reply:
[https://github.com/opiproject/sztp/pull/81](https://github.com/opiproject/sztp/pull/81)